libgalois/include/katana/analytics/matrix_completion/matrix_completion.h

```diff
@@ -0,0 +1,148 @@
+      bool use_det_init, Step learning_rate_function)
+      : Plan(architecture),
+        algorithm_(algorithm),
+        learning_rate_(learning_rate),
+        decay_rate_(decay_rate),
+        lambda_(lambda),
+        tolerance_(tolerance),
+        use_same_latent_vector_(use_same_latent_vector),
+        max_updates_(max_updates),
+        updates_per_edge_(updates_per_edge),
+        fixed_rounds_(fixed_rounds),
+        use_exact_error_(use_exact_error),
+        use_det_init_(use_det_init),
+        learning_rate_function_(learning_rate_function) {}
+
+public:
+  MatrixCompletionPlan()
+      : MatrixCompletionPlan{
+            kCPU,
+            kSGDByItems,
+            kDefaultLearningRate,
+            kDefaultDecayRate,
+            kDefaultLambda,
+            kDefaultTolerance,
+            kDefaultUseSameLatentVector,
+            kDefaultMaxUpdates,
+            kDefaultUpdatesPerEdge,
+            kDefaultFixedRounds,
+            kDefaultUseExactError,
+            kDefaultUseDetInit,
+            kDefaultLearningRateFunction} {}
+
+  Algorithm algorithm() const { return algorithm_; }
+  double learningRate() const { return learning_rate_; }
+  double decayRate() const { return decay_rate_; }
+  double lambda() const { return lambda_; }
+  double tolerance() const { return tolerance_; }
+  bool useSameLatentVector() const { return use_same_latent_vector_; }
+  uint32_t maxUpdates() const { return max_updates_; }
+  uint32_t updatesPerEdge() const { return updates_per_edge_; }
+  uint32_t fixedRounds() const { return fixed_rounds_; }
+  bool useExactError() const { return use_exact_error_; }
+  bool useDetInit() const { return use_det_init_; }
+  Step learningRateFunction() const { return learning_rate_function_; }
+
+  static MatrixCompletionPlan SGDByItems(
+      double learning_rate = kDefaultLearningRate,
+      double decay_rate = kDefaultDecayRate, double lambda = kDefaultLambda,
+      double tolerance = kDefaultTolerance,
+      bool use_same_latent_vector = kDefaultUseSameLatentVector,
+      uint32_t max_updates = kDefaultMaxUpdates,
+      uint32_t updates_per_edge = kDefaultUpdatesPerEdge,
+      uint32_t fixed_rounds = kDefaultFixedRounds,
+      bool use_exact_error = kDefaultUseExactError,
+      bool use_det_init = kDefaultUseDetInit,
+      Step learning_rate_function = kDefaultLearningRateFunction) {
+    return {
+        kCPU,
+        kSGDByItems,
+        learning_rate,
+        decay_rate,
+        lambda,
+        tolerance,
+        use_same_latent_vector,
+        max_updates,
+        updates_per_edge,
+        fixed_rounds,
+        use_exact_error,
+        use_det_init,
+        learning_rate_function};
+  }
+};
+
+struct KATANA_EXPORT MatrixCompletionStatistics {
+  /// The number of iterations
+  uint32_t num_iterations;
+
+  // RMSE of the last iteration
+  double last_rmse;
+
+  /// Print the statistics in a human readable form.
+  void Print(std::ostream& os = std::cout) const;
+};
```

Comment:
This isn't what the statistics structures are for. This may be useful information. But the statistics for an algorithm are supposed to provide information about the graph not the run of the algorithm. If this is important info you could have a struct returned by `MatrixCompletion` but don't call it `Statistics` because it doesn't match the convention set by other algorithms.

Reply:
Thanks for the clarification. I understand that the `Statistics` use the `compute` function to compute information about the graph after running the algorithm. @gurbinder533, since we discussed this before, could you please comment on this?

Reply:
I tend to have similar "Please follow the conventions" comments on @gurbinder533's work too. ;-)

Reply:
You are right that the conventions of other algorithms needs to be followed. I talked with Gurbinder and since we don't need any statistics for the case of matrix completion right now, I just removed the `Statistics` class for now.